myhpi/templates/footer.html

```diff
@@ -12,9 +12,16 @@
     {% endfor %}
     <div class="footer-category">
       <h2>{% translate "Language" %}</h2>
-        {% for translation in page.get_translations %}
-            <a href="{{ translation.url }}">{{ translation.locale }}</a>
-        {% endfor %}
+      {% for translation in page.get_translations %}
+      <form id="language-form-{{ translation.locale }}" action="{% url 'set_language' %}" method="post" style="display: none;">
+        {% csrf_token %}
+        <input name="next" type="hidden" value="{{ redirect_to }}">
```

Comment:
where is redirect_to coming from? shouldn't that be translation.url?

Reply:
It was always empty, so it seemed to work without setting next, but I applied your suggestion now.